strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+    function can plot just the input graph or the graph with a specified subgraph highlighted.
+    The function uses the standard colour theme of green nodes, grey edges, and red highlighted
+    subgraph.
+
+    **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> fig = plot(graph, [0, 1, 2, 3])
+        >>> fig.show()
+
+        Args:
+            graph (nx.Graph): input graph
+            subgraph (list): list of nodes comprising the subgraph to highlight
+            size (dict): size of the plot
+
+        Returns:
+             plot.ly graph object
+        """
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode='markers',
+        hoverinfo='text',
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2)
+    )
+
+    g_edges = go.Scatter(
+        **edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo='none',
+        mode='lines'
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
+
+    layout = go.Layout(showlegend=False, hovermode='closest',
+                       xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       margin=dict(b=0, l=0, r=0, t=25),
+                       height=size,
+                       width=size,
+                       plot_bgcolor='#ffffff'
+                       )
+
+    if subgraph:
+
+        s_edges = go.Scatter(
+            **edge_coords(s, l),
+            line=dict(width=2, color=subgraph_edge_colour),
+            hoverinfo='none',
+            mode='lines'
+        )
+
+        s_nodes = go.Scatter(
+            **_node_coords(s, l),
+            mode='markers',
+            hoverinfo='text',
+            marker=dict(color=subgraph_node_colour, size=subgraph_node_size, line_width=2)
+        )
+
+        s_nodes.text = [str(i) for i in s.nodes()]
+
+        f = go.Figure(data=[g_edges, s_edges, g_nodes, s_nodes], layout=layout)
+
+    else:
+        f = go.Figure(data=[g_edges, g_nodes], layout=layout)
+
+    return f
+
+
+def plot_subgraph(subgraph: nx.Graph, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input subgraph.
+
+        The graph layout is fixed to be the Kamada-Kawai layout with an aspect ratio of 1:1.
+        This function uses the subgraph colour theme of red nodes and edges.
+
+        **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> subgraph = graph.subgraph([0, 1, 2, 3])
+        >>> fig = plot_subgraph(subgraph)
+        >>> fig.show()
+
+            Args:
+                subgraph (nx.Graph): input subgraph
+                size (dict): size of the plot
+
+            Returns:
+                 plot.ly graph object
+            """
+
+    l = nx.kamada_kawai_layout(subgraph)
+
+    g_edges = go.Scatter(
```

Comment:
Would it make sense for a `_add_graph_to_plot` private function to do these lines, which I think are more or less repeated above

Reply:
I tried to do this but in the end concluded it's not worth it for two main reasons:
1. The format is different for plotting a graph and a subgraph, so the `_add_graph_to_plot` function needs to acommodate both options. When doing so, we don't save that many lines of codes
2. Somehow the formatting of the plot is different and not as nice. For example, the edges of the graph now appear on top of the nodes, instead of hidden behind.

All these things could be fixed with more work, but I don't think it's worth it

Reply:
Ok, agreed in person to drop this